addons/stock/models/stock_picking.py

```diff
@@ -194,6 +194,18 @@ def write(self, vals):
                         'prefix': vals['sequence_code'], 'padding': 5,
                         'company_id': picking_type.env.company.id,
                     })
+        if 'reservation_method' in vals:
+            if vals['reservation_method'] == 'by_date':
+                for picking_type in self:
+                    moves = self.env['stock.move'].search([('picking_type_id', '=', picking_type.id), ('state', 'in', ('draft', 'confirmed', 'waiting', 'partially_available'))])
+                    common_days = picking_type.reservation_days_before
+                    priority_days = picking_type.reservation_days_before_priority
+                    for move in moves:
+                        move.reservation_date = fields.Date.to_date(move.date) - timedelta(days=priority_days if move.priority == '1' else common_days)
+            else:
+                moves = self.env['stock.move'].search([('picking_type_id', 'in', self.ids), ('state', 'not in', ('assigned', 'done', 'cancel'))])
+                moves.reservation_date = False
```

Comment:
Do you think it's worth it if we also do a check to make sure that the `vals['reservation_method'] != the current reservation_method`? So that we avoid unnecessarily search/writes to the `reservation_date` in that case?

Reply:
@ticodoo I think you are right ! You can r- imo :) Also, we can use a read_group instead of doing a search in the loop to optimize the computation when it has to be done !

Reply:
Hello @ticodoo, I have updated the diff and tested the flow on a large DB, tell me what you think :)